components/user_profile/index.ts

```diff
@@ -5,14 +5,16 @@ import {connect} from 'react-redux';
 
 import {getUser, makeGetDisplayName} from 'mattermost-redux/selectors/entities/users';
 
-import UserProfile from './user_profile';
+import {GlobalState} from 'mattermost-redux/types/store';
+
+import UserProfile, {UserProfileProps} from './user_profile';
 
 function makeMapStateToProps() {
     const getDisplayName = makeGetDisplayName();
 
-    return (state, ownProps) => {
+    return (state: GlobalState, ownProps: UserProfileProps) => {
```

Comment:
Technically the `UserProfile` component shouldn't accept all of the fields as props that the `UserProfile` object provides. I think we should just provide our own `ownProps` type specific to this file in the connector. 2/5.

Reply:
I think this is fine if it accepts all the props of `UserProfile` since `connect` returns a new component that accepts the props of `UserProfile` plus whatever we use here

Reply:
The component doesn't though, it only accepts a subset of properties from `UserProfile`. There are a couple that line up, but not across the board.

Reply:
What about doing the following:

```typescript
type OwnProps = {
    userId: string,
}
```

and then replace `UserProfileProps` with `OwnProps`?

Any better name idea for `OwnProps`?

Reply:
Actually ended up doing the above. Let me know what you think. I am happy to change approach and names based on your feedback!

Reply:
Yep, that's perfect! Thanks @lmammino!